target/scripts/startup/setup-stack.sh

```diff
@@ -437,10 +437,17 @@ function _setup_ldap
 
   declare -A _dovecot_ldap_mapping
 
+  # Add protocol to LDAP_SERVER_HOST so that we can use dovecot's "uris" option:
+  # https://doc.dovecot.org/configuration_manual/authentication/ldap/
+  dovecot_ldap_server_uri="${LDAP_SERVER_HOST}"
+  if ! echo "${dovecot_ldap_server_uri}" | grep -F '://' >/dev/null; then
+    dovecot_ldap_server_uri="ldap://${dovecot_ldap_server_uri}"
+  fi
+
   _dovecot_ldap_mapping["DOVECOT_BASE"]="${DOVECOT_BASE:="${LDAP_SEARCH_BASE}"}"
   _dovecot_ldap_mapping["DOVECOT_DN"]="${DOVECOT_DN:="${LDAP_BIND_DN}"}"
   _dovecot_ldap_mapping["DOVECOT_DNPASS"]="${DOVECOT_DNPASS:="${LDAP_BIND_PW}"}"
-  _dovecot_ldap_mapping["DOVECOT_HOSTS"]="${DOVECOT_HOSTS:="${LDAP_SERVER_HOST}"}"
+  _dovecot_ldap_mapping["DOVECOT_URIS"]="${DOVECOT_URIS:="${dovecot_ldap_server_uri}"}"
```

Comment:
NOTE: This note is for maintainers, and should not affect this PR.

This whole section was never worked on, therefore the old style is present. This should not be a problem as of now, but we should re-write this in the future to have consistent styling.

Reply:
I just noticed that this deprecates the DOVECOT_HOSTS variable, I'll change it so that it's still supported?

Reply:
> I just noticed that this deprecates the DOVECOT_HOSTS variable, I'll change it so that it's still supported?
> Edit: I markeed DOVECOT_HOSTS as deprecated and use it as a fallback value for DOVECOT_URIS now.

Exactly. We cannot get rid of this in an instant as we need to ensure backwards compatibility in a major version. When 10.0.0 is release, we can change this. The deprecation notice is the absolutely right way.